Automattic/simplenote-android
# Add support for nested checklists

Depends on #608 

Markdown checklists can be nested, so I've added support for that:

<img width="178" alt="screen shot 2019-01-18 at 9 54 30 am" src="https://user-images.githubusercontent.com/789137/51404163-739e8380-1b07-11e9-9c46-485affa16011.png">

**To Test**
* Create a checklist at any position on a line. It should insert at the proper position.
* Pressing enter should 'autobullet' a new checklist item on the next line at the same position.
* Selecting multiple lines and/or one line of text and then inserting/removing a checklist should still work as expected.
